boogie-routes/src/main/java/org/mitre/tdp/boogie/alg/chooser/GraphicalRouteChooser.java

```diff
@@ -89,6 +84,66 @@ public List<ResolvedLeg> chooseRoute(List<ResolvedTokens> resolvedTokens) {
         .toList();
   }
 
+  private List<ResolvedTokens> filterLinkableTokens(List<ResolvedTokens> resolvedTokens, Set<Leg> resolvedEntryPoints, Set<Leg> resolvedExitPoints) {
+
+    List<String> entryNames = resolvedEntryPoints.stream()
+        .map(Leg::associatedFix)
+        .flatMap(Optional::stream)
+        .map(Fix::fixIdentifier)
+        .toList();
+
+    List<String> exitNames = resolvedExitPoints.stream()
+        .map(Leg::associatedFix)
+        .flatMap(Optional::stream)
+        .map(Fix::fixIdentifier)
+        .toList();
+
+    return resolvedTokens.stream()
+        .map(tokens -> removeUnlinkableSids(tokens, entryNames))
+        .flatMap(Optional::stream)
+        .map(tokens -> removeUnlinkableStars(tokens, exitNames))
+        .flatMap(Optional::stream)
+        .filter(tokens -> !tokens.resolvedTokens().isEmpty())
+        .toList();
+  }
+
+  private Optional<ResolvedTokens> removeUnlinkableStars(ResolvedTokens tokens, List<String> entryNames) {
+    List<ResolvedToken> filteredTokens = tokens.resolvedTokens().stream()
+        .filter(token -> !unlinkableStar(token, entryNames))
+        .toList();
+    return filteredTokens.isEmpty()
+        ? Optional.empty()
+        : Optional.of(new ResolvedTokens(tokens.routeToken(), filteredTokens));
+  }
+
+  private boolean unlinkableStar(ResolvedToken token, List<String> entryNames) {
+    if (!(token instanceof ResolvedToken.StarEnrouteCommon || token instanceof ResolvedToken.StarRunway)) {
```

Comment:
I'm not very happy about the `instanceof` checks, but accessing the kind of token is very difficult